pkgs/development/haskell-modules/configuration-ghc-7.10.x.nix

```diff
@@ -80,6 +91,30 @@ self: super: {
   # Test suite fails in "/tokens_bytestring_unicode.g.bin".
   alex = dontCheck super.alex;
 
+  haddock-library = overrideCabal super.haddock-library (drv: {
```

Comment:
That should be fixed with haddock-library which will be [released](https://github.com/haskell/haddock/issues/366) together with ghc 7.10. I'm not sure if it is worth adding workarounds for the expected two or three weeks if everything goes according to plan